scrapy_zyte_api/handler.py

```diff
@@ -32,6 +33,20 @@ def __init__(
         self._session = create_session()
         self._encoding = "utf-8"
 
+    @classmethod
+    def from_crawler(cls, crawler):
+        zyte_api_key = crawler.settings.get("ZYTE_API_KEY") or os.getenv("ZYTE_API_KEY")
+        if not zyte_api_key:
+            logger.warning(
+                "'ZYTE_API_KEY' must be set in the spider settings or env var "
+                "in order for ScrapyZyteAPIDownloadHandler to work."
+            )
+            raise NotConfigured
+
+        logger.info(f"Using Zyte API Key: {zyte_api_key[:7]}")
```

Comment:
Why is it cut? Is it for security reasons? Would it make sense to add "*" for the rest of the key?

Reply:
Yep, following the convention of `scrapy-zyte-smartproxy`: https://github.com/scrapy-plugins/scrapy-zyte-smartproxy/blob/master/scrapy_zyte_smartproxy/middleware.py#L85-L90.

I'm not quite sure about having **"*"** for the rest as it could pollute the logs a bit. Though I think having the first few chars would be enough to have a good reference on the API that's currently set.

Reply:
👍 I think if wre want to change it, it'd be better to change it in both places; it's out of scope for this PR.